### PR TITLE
ci: avoid future mistake

### PR DESCRIPTION
**Pull request by jorgenengelsen**
## Why is this pull request needed?

This pull request is needed because the wrong typescript client got pushed by accident. Avoid this mistake in the future

## What does this pull request change?

Check that the correct openapi spec is used to generate the typescript client

### CODE REVIEW

File: api/pyproject.toml [hunk 1]
```diff
@@ -96,8 +96,12 @@ ignore = [
 
 [tool.ruff.per-file-ignores]
 "__init__.py" = ["E402"]  # Ignore `E402` (import violations) in all `__init__.py` files
-"locustfile.py" = ["B012"]  # B012 `break` inside `finally` blocks cause exceptions to be silenced
```
Comment (sebastianvitterso) [inline api/pyproject.toml:99]: Feels unrelated?

File: api/pyproject.toml [hunk 2]
```diff
@@ -96,8 +96,12 @@ ignore = [
...
 
 [tool.codespell]
 skip = "*.lock,*.cjs"
 ignore-words-list = "ignored-word"
+
+[tool.pytest]
+markers =[
+    "integration: mark a test as integration test."
+]
```
Comment (sebastianvitterso) [inline api/pyproject.toml:107]: Feels unrelated?

Comment (sebastianvitterso) [reply api/pyproject.toml:107]: But fair enough.

File: web/generate-api-typescript-client-pre-commit.sh [hunk 3]
```diff
@@ -14,6 +14,36 @@ if [ ${#CHANGED_API_FILES[@]} -eq 0 ]; then
...
+      echo "The openapi specification found at localhost:5000 ('$NAME') does not seem to belong to 'Template FastAPI React'"
+      exit 1
+    else
+      cd web
+      yarn openapi -i http://localhost:5000/openapi.json -o "$PWD/src/api/generated"
+      echo "API Client successfully generated"
+    fi
+
+    # Clean up the temporary file
+    rm "$TEMP_FILE"
```
Comment (sebastianvitterso) [inline web/generate-api-typescript-client-pre-commit.sh:48]: I must admit this feels like overengineering it, but I might be wrong.

I can't find any issues in the code flow, so if this is desired functionality, it's fine.

**Review by sebastianvitterso — APPROVED** [review]

File: web/generate-api-typescript-client-pre-commit.sh (hunk 3 above)
Comment (pbullhove) [reply web/generate-api-typescript-client-pre-commit.sh:48]: Seems fair from my point of view.

File: api/pyproject.toml (hunk 1 above)
Comment (jorgenengelsen) [reply api/pyproject.toml:99]: I guess it is 🙌
